### DISCUSSION TITLE
Add create button in plugin manage section.

### DISCUSSION THREAD

**Discussion by LaocheXe** [e107inc/e107#5021] [Ideas]
Creating a plugin with the plugin builder, if someone has a lot of database tables, the admin_config.php has a punch of Manage(list), and Create(create). It would be nice if we can remove the create, and just add it as a button on the list(Manage) page. Something simple like what the create page has with the Create/Cancel butt.

Example: 
What it looks like:
![Screenshot 2023-06-23 155522](https://github.com/e107inc/e107/assets/5818022/dd8e45b6-129a-4928-b899-e69904abcde3)

What I want it to look like:
![example1](https://github.com/e107inc/e107/assets/5818022/6ff6af91-d773-4a37-a7ce-0cc0c6b7ab67)

I would try to just do it myself, but I have no clue where to look to do so, I would make a custom page with my own button, but again - I don't know where the template is for admin config's for plugins.

**Comment by Jimmi08** [comment]
Hi, I asked about this years ago...   I think that @rica-carv found a workaround I am using

The only way how I was able to do it: (It is just a screenshot from some plugin I have online - to see position)
![image](https://github.com/e107inc/e107/assets/5429548/cfddf3de-ed31-45b8-96d9-5b32ff3a8752)

put this in init()
$this->postFilterMarkup = $this->AddButton();

and add method - just an example from searching github online, I have a better way somewhere (when I find it, I will add it here)

```
	  function AddButton()
	  {
      $text  = "</fieldset></form><div class='e-container'>
      <table id='.$this->id.' style='".ADMIN_WIDTH."' class='table adminlist table-striped'>";
      $text .=  
      '<a href="admin_config.php?mode=main&action=create"  
      class="btn batch e-hide-if-js btn-primary"><span>Add Plan</span></a>';
      $text .= "</td></tr></table></div><form><fieldset>";
      return $text;
    }
```

You can/could find this in many of my plugins. 

About the admin menu...   in my plugins, I always separate the menu and admin UI part. It is not recommended and I create a folder admin.  One of the main reasons is that I don't want to admin_config.php replaced by using a plugin builder repeatedly (it is needed with complex plugins).  During that time I had a lot of versions of how to do it...  my questions about this weren't answered.   With each plugin, it was better and better. Unfortunately, I lost my backup, the only available sources are left on github.

or look in "vstore" plugin for how to create a separate admin page for each admin menu item.  Maybe you will get it. I didn't at that time, so I am using my way,

**Reply by Jimmi08** [reply]
I just remembered how they broke my plugins with this change:  https://github.com/e107inc/e107/commit/77baec4f063eb2a0272553f9f263db4dd0b74e8a?diff=unified

without any callback or warning.  And this was the reason why I stopped to used this and I split complex plugin into more than one admin area or more plugins.   Bad memory :)  So if you search github, search for both names.

**Reply by LaocheXe** [reply]
This works, just wish I knew how to get it below the table, and not above it. Having it display under the search function doesn't look that good to me.
